tests/test_parallelreduce.py

```diff
@@ -40,5 +44,53 @@ def test_add_squares(self):
         self.assertEqual(expected_result, result)
 
 
+@pk.workload
+class SquareSumFloat:
+    def __init__(self, n):
+        self.N: int = n
+        self.total: pk.double = 0
+
+    @pk.main
+    def run(self):
+        self.total = pk.parallel_reduce(self.N, self.squaresum)
+
+    @pk.workunit
+    def squaresum(self, i: float, acc: pk.Acc[pk.double]):
+        acc += i * i
+
+
+@pk.workload
+class SquareSumInt:
```

Comment:
Is there a way for me to "parametrize" the workloads by type rather than copy/pasting and using slightly different types for testing like this? Seems tricky maybe because the type hints can't really be substituted in?